### PR TITLE
statistics: avoid stats meta full load after table analysis

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/57631

Problem Summary:

See more at https://github.com/pingcap/tidb/issues/57631#issuecomment-2499607905

We need to support updating the stats of only a few tables within the update function. This will help reduce the function's duration after the table analysis is completed.

The analysis duration is stable:
```sql
mysql> analyze table sbtest111.sbtest2;
Query OK, 0 rows affected, 10 warnings (6.22 sec)

mysql> analyze table sbtest111.sbtest2;
Query OK, 0 rows affected, 10 warnings (6.74 sec)

mysql> analyze table sbtest111.sbtest2;
Query OK, 0 rows affected, 10 warnings (6.12 sec)
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by Rustin170506** [comment]
Test result:
image


The analysis duration is stable:
```sql
mysql> analyze table sbtest111.sbtest2;
Query OK, 0 rows affected, 10 warnings (6.22 sec)

mysql> analyze table sbtest111.sbtest2;
Query OK, 0 rows affected, 10 warnings (6.74 sec)

mysql> analyze table sbtest111.sbtest2;
Query OK, 0 rows affected, 10 warnings (6.12 sec)
```

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: pkg/statistics/handle/storage/read.go [hunk 1]
```diff
@@ -782,7 +794,14 @@ func loadNeededIndexHistograms(sctx sessionctx.Context, is infoschema.InfoSchema
 		tbl.LastAnalyzeVersion = max(tbl.LastAnalyzeVersion, idxHist.LastUpdateVersion)
 	}
 	tbl.SetIdx(idx.ID, idxHist)
-	statsHandle.UpdateStatsCache([]*statistics.Table{tbl}, nil)
+	tbl.LastAnalyzeVersion = max(tbl.LastAnalyzeVersion, idxHist.LastUpdateVersion)
```
Comment (time-and-fate) [inline pkg/statistics/handle/storage/read.go:797]: I think this line should be removed.

**Review by time-and-fate — COMMENTED** [review]
Others LGTM.

File: pkg/statistics/handle/storage/read.go (hunk 1 above)
Comment (Rustin170506) [reply pkg/statistics/handle/storage/read.go]: Oh, it was accidentally introduced while I was resolving conflicts.

Comment (Rustin170506) [reply pkg/statistics/handle/storage/read.go]: Removed.

File: pkg/statistics/handle/types/interfaces.go [hunk 2]
```diff
@@ -195,6 +195,22 @@ type StatsAnalyze interface {
 	Close()
 }
 
+// CacheUpdate encapsulates changes to be made to the stats cache
+type CacheUpdate struct {
+	Added   []*statistics.Table
```
Comment (time-and-fate) [inline pkg/statistics/handle/types/interfaces.go:200]: Maybe `Updated` is a better name?

File: pkg/statistics/handle/storage/update.go [hunk 3]
```diff
@@ -211,7 +211,13 @@ func removeExtendedStatsItem(statsCache types.StatsCache,
 	}
 	newTbl := tbl.Copy()
 	delete(newTbl.ExtendedStats.Stats, statsName)
-	statsCache.UpdateStatsCache([]*statistics.Table{newTbl}, nil)
+	statsCache.UpdateStatsCache(types.CacheUpdate{
+		Added:   []*statistics.Table{newTbl},
+		Deleted: []int64{},
+		Options: types.UpdateOptions{
+			SkipMoveForward: false,
+		},
```
Comment (time-and-fate) [inline pkg/statistics/handle/storage/update.go:219]: What about removing these two fields for existing usage without changes?

**Review by time-and-fate — APPROVED** [review]
/hold

File: pkg/statistics/handle/types/interfaces.go (hunk 2 above)
Comment (Rustin170506) [reply pkg/statistics/handle/types/interfaces.go]: Good point!

File: pkg/statistics/handle/storage/update.go (hunk 3 above)
Comment (Rustin170506) [reply pkg/statistics/handle/storage/update.go]: Removed.

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Pull request merged**